### DISCUSSION TITLE
Dynamic links

### DISCUSSION THREAD

**Discussion by mint-ws** [spatie/laravel-onboard#36] [Q&A] [answered]
New to using service providers so not sure if it's possible to use dynamic data and/or route names in the  ```link()```.

For example, how do I point to a route name, or just a hard-coded url, that requires a parameter from the current model instance?

Any advice greatly appreciated.

**Comment by ap1969** [comment]
Doesn't seem to be possible, from what I can make out. Just came here to find out the same as I want to pass a dynamic ID to the end of the URL.

**Comment by ap1969** [comment, marked as answer]
Found a work-around. Create a web route with the dynamic elements and link to that, eg
```
Onboard::addStep('Assign Activity To Checklist')
            ->link("/checklist-assign")
            ->cta('Edit Checklist')
            ->completeIf(function () {
                $user = auth()->user();
                if (! $user || ! $user->checklists()->exists()) {
                    return false;
                }
                return $user->checklists()->first()->activities()->count() > 0;
            });
```
```
Route::get('/checklist-assign', function () {
    return redirect('/admin/checklists/' . auth()->user()->checklists()->first()->id . '/edit');
})->middleware('web');
```

**Reply by mint-ws** [reply]
Hi @ap1969 thanks for the work-around.
Bookmarked for next time as I had to roll my own after the long silence on this question.
